Fix SwitchCell not compliant with best accessibility practices

dawidpieper opened the pull request:
### Description of Change

For screenreader users on iOS and Android, SwitchCell is reported as two objects: one being the label, and one the switch itself.
It's always the best practice to combine them into one navigation element, being a labeled switch.

### Issues Fixed

Fixes #6683

dawidpieper committed:
Fix #6683

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=6899) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=6899) before we can accept your contribution.<br/><br/>:x: dawidpieper [sign now](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=6899)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/maui?pullRequest=6899) it.</sub>

dawidpieper committed:
Make calls compliant with Android API
Tested on Android 11 using TalkBack and Commentary Screenreader, seems to work fine

dawidpieper commented:
Seems to work, tested on Android 11 and iOS 15.4.1.
Ready for review.

dawidpieper commented:
Note, PR incorrectly marked as closing # 6648, in fact closing # 6683.

rmarinho commented:
/azp run

PureWeen committed:
Merge branch 'main' into pr/6899

PureWeen commented on src/Controls/src/Core/Compatibility/Handlers/ListView/Android/BaseCellView.cs:
```diff
@@ -116,18 +119,26 @@ Element INativeElementView.Element
+				_accessoryViewOverrideAsAccessibilityElement = overrideAsAccessibilityElement;
+				if (overrideAsAccessibilityElement)
+					_mainText.LabelFor = view.Id;
+					_mainText.ImportantForAccessibility = ImportantForAccessibility.No;
```
To @samhouts comment here

https://github.com/dotnet/maui/issues/6683#issuecomment-1144265619

I don't think this is correct behavior to set this to No

This makes it so accessibility only selects the switch and never the entire listview row. 

From testing inside the settings app the swipe behavior goes

- <Reads out description of list item cell> "double tap to activate"
- swipe right
- selects switch

It's important to have that first "double tap to activate" because the behavior might be that you want to double tap the cell and still select it.
